src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1530,6 +1544,99 @@ def _get_linux_multicontainer_encoded_config_from_file(file_name):
     # Decode base64 encoded byte array into string
     return b64encode(config_file_bytes).decode('utf-8')
 
+def update_deployment_configs(cmd, resource_group_name, name,
+                              deployment_storage_name=None,
+                              deployment_storage_container_name=None, deployment_storage_auth_type=None,
+                              deployment_storage_auth_value=None, slot=None):
+    
+    # NameValuePair = cmd.get_models('NameValuePair')
+    Site, SiteConfig, NameValuePair, AppSettings = cmd.get_models('Site', 'SiteConfig', 'NameValuePair', 'AppSettings')
+    
+    logger.warning("checking the flex app. Params: deployment_storage_container_name: %s", deployment_storage_container_name)
+    is_flex = is_flex_functionapp(cmd.cli_ctx, resource_group_name, name)
+
+    if not is_flex:
+        raise ValidationError("This command is only valid for Azure Functions on the FlexConsumption plan.")
+
+    logger.warning("deployment_storage_name: %s deployment_storage_container_name: %s", deployment_storage_name, deployment_storage_container_name)
+    logger.warning("deployment_storage_auth_type: %s deployment_storage_auth_value: %s", deployment_storage_auth_type, deployment_storage_auth_value)
+
+    if deployment_storage_name == None or deployment_storage_container_name == None:
+        raise ValidationError("--deployment-storage-name and --deployment-storage-container-name are required.")
+    
+    client = web_client_factory(cmd.cli_ctx)
+    functionapp = client.web_apps.get(resource_group_name, name)
+    
+    if functionapp is not None:
+        logger.warning("functionapp %s in %s found", functionapp.name, functionapp.location)
+
+    deployment_storage = _validate_and_get_deployment_storage(cmd.cli_ctx, resource_group_name, deployment_storage_name)
+    deployment_storage_container = _get_deployment_storage_container(cmd, resource_group_name, deployment_storage_name, deployment_storage_container_name)
+    deployment_storage_container_name = deployment_storage_container.name
+    deployment_config_storage_value = getattr(deployment_storage.primary_endpoints, 'blob') + deployment_storage_container_name
+    deployment_storage_auth_type = deployment_storage_auth_type or 'systemAssignedIdentity'
+
+    if deployment_storage_auth_value and deployment_storage_auth_type != 'userAssignedIdentity':
+        raise ArgumentUsageError(
+            '--deployment-storage-auth-value is only a valid input for --deployment-storage-auth-type set to userAssignedIdentity. '
+            'Please try again with --deployment-storage-auth-type set to userAssignedIdentity.'
+        )
+
+    if deployment_storage_auth_type == 'userAssignedIdentity':
```

Comment:
We need to make sure that the customer passes the deployment_storage_auth_value when userAssignedIdentity is passed. We are only going to validate the value but will not create a user assigned identity for them.

Reply:
I have added the validation.